### PR TITLE
Make it clearer that the `iterator` method will return a fresh Iterator at each call

### PR DESCRIPTION
**Pull request by julienrf**
Also removes the cycle between `Iterator` and `IterableOnce`.

### CODE REVIEW

File: src/test/scala/strawman/collection/test/Test.scala [hunk 1]
```diff
@@ -121,9 +121,9 @@ class StrawmanTest {
     val ys7: String = xs7
     val xs8 = xs.drop(2)
     val ys8: String = xs8
-    val xs9 = xs.map(_ + 1)  // !!! need a language change to make this work without the : Char
+    val xs9 = xs.map(_ + 1)
```
Comment (DarkDimius) [inline src/test/scala/strawman/collection/test/Test.scala]: why was the comment removed?

Comment (julienrf) [inline src/test/scala/strawman/collection/test/Test.scala]: Because the type ascription has already gone.

Comment (DarkDimius) [inline src/test/scala/strawman/collection/test/Test.scala]: Ah, i see. Scalac has already introduced same overload resolution rule as dotty did. Thanks!

**Comment by odersky** [comment]
LGTM 👍

**Pull request merged**